// steel-sdk/resources/extensions.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../resource.mjs";
import { isRequestOptions } from "../core.mjs";
import * as Core from "../core.mjs";
export class Extensions extends APIResource {
    update(extensionId, body = {}, options) {
        if (isRequestOptions(body)) {
            return this.update(extensionId, {}, body);
        }
        return this._client.put(`/v1/extensions/${extensionId}`, Core.multipartFormRequestOptions({ body, ...options }));
    }
    /**
     * List all extensions for the organization
     */
    list(options) {
        return this._client.get('/v1/extensions', options);
    }
    /**
     * Delete an extension by ID
     */
    delete(extensionId, options) {
        return this._client.delete(`/v1/extensions/${extensionId}`, options);
    }
    /**
     * Delete all extensions for the organization
     */
    deleteAll(options) {
        return this._client.delete('/v1/extensions', options);
    }
    /**
     * Download an extension file by extension ID
     */
    download(extensionId, options) {
        return this._client.get(`/v1/extensions/${extensionId}`, options);
    }
    upload(body = {}, options) {
        if (isRequestOptions(body)) {
            return this.upload({}, body);
        }
        return this._client.post('/v1/extensions', Core.multipartFormRequestOptions({ body, ...options }));
    }
}
