// steel-sdk/resources/files.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Files = void 0;
const resource_1 = require("../resource.js");
const Core = __importStar(require("../core.js"));
class Files extends resource_1.APIResource {
    /**
     * List all global files for the organization in descending order.
     */
    list(options) {
        return this._client.get('/v1/files', options);
    }
    /**
     * Delete a file from global storage
     */
    delete(path_, options) {
        return this._client.delete(`/v1/files/${path_}`, {
            ...options,
            headers: { Accept: '*/*', ...options?.headers },
        });
    }
    /**
     * Download a file from global storage
     */
    download(path_, options) {
        return this._client.get(`/v1/files/${path_}`, {
            ...options,
            headers: { Accept: 'application/octet-stream', ...options?.headers },
            __binaryResponse: true,
        });
    }
    /**
     * Uploads a file to global storage via `multipart/form-data` with a `file` field
     * that accepts either binary data or a URL string to download from, and an
     * optional `path` field for the file storage path.
     */
    upload(body, options) {
        return this._client.post('/v1/files', Core.multipartFormRequestOptions({ body, ...options }));
    }
}
exports.Files = Files;
