// steel-sdk/resources/files.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../resource.mjs";
import * as Core from "../core.mjs";
export class Files extends APIResource {
    /**
     * List all global files for the organization in descending order.
     */
    list(options) {
        return this._client.get('/v1/files', options);
    }
    /**
     * Delete a file from global storage
     */
    delete(path_, options) {
        return this._client.delete(`/v1/files/${path_}`, {
            ...options,
            headers: { Accept: '*/*', ...options?.headers },
        });
    }
    /**
     * Download a file from global storage
     */
    download(path_, options) {
        return this._client.get(`/v1/files/${path_}`, {
            ...options,
            headers: { Accept: 'application/octet-stream', ...options?.headers },
            __binaryResponse: true,
        });
    }
    /**
     * Uploads a file to global storage via `multipart/form-data` with a `file` field
     * that accepts either binary data or a URL string to download from, and an
     * optional `path` field for the file storage path.
     */
    upload(body, options) {
        return this._client.post('/v1/files', Core.multipartFormRequestOptions({ body, ...options }));
    }
}
