// steel-sdk/index.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var _Steel_instances, _a, _Steel_baseURLOverridden;
import * as Core from "./core.mjs";
import * as Errors from "./error.mjs";
import * as Pagination from "./pagination.mjs";
import * as Uploads from "./uploads.mjs";
import * as API from "./resources/index.mjs";
import { Credentials, } from "./resources/credentials.mjs";
import { Extensions, } from "./resources/extensions.mjs";
import { Files } from "./resources/files.mjs";
import { Profiles, } from "./resources/profiles.mjs";
import { Sessions, SessionslistSessionsSessionsCursor, } from "./resources/sessions/sessions.mjs";
/**
 * API Client for interfacing with the Steel API.
 */
export class Steel extends Core.APIClient {
    /**
     * API Client for interfacing with the Steel API.
     *
     * @param {string | null | undefined} [opts.steelAPIKey=process.env['STEEL_API_KEY'] ?? null]
     * @param {string} [opts.baseURL=process.env['STEEL_BASE_URL'] ?? https://api.steel.dev] - Override the default base URL for the API.
     * @param {number} [opts.timeout=1 minute] - The maximum amount of time (in milliseconds) the client will wait for a response before timing out.
     * @param {number} [opts.httpAgent] - An HTTP agent used to manage HTTP(s) connections.
     * @param {Core.Fetch} [opts.fetch] - Specify a custom `fetch` function implementation.
     * @param {number} [opts.maxRetries=2] - The maximum number of times the client will retry a request.
     * @param {Core.Headers} opts.defaultHeaders - Default headers to include with every request to the API.
     * @param {Core.DefaultQuery} opts.defaultQuery - Default query parameters to include with every request to the API.
     */
    constructor({ baseURL = Core.readEnv('STEEL_BASE_URL'), steelAPIKey = Core.readEnv('STEEL_API_KEY') ?? null, ...opts } = {}) {
        const options = {
            steelAPIKey,
            ...opts,
            baseURL: baseURL || `https://api.steel.dev`,
        };
        super({
            baseURL: options.baseURL,
            baseURLOverridden: baseURL ? baseURL !== 'https://api.steel.dev' : false,
            timeout: options.timeout ?? 60000 /* 1 minute */,
            httpAgent: options.httpAgent,
            maxRetries: options.maxRetries,
            fetch: options.fetch,
        });
        _Steel_instances.add(this);
        this.credentials = new API.Credentials(this);
        this.files = new API.Files(this);
        this.sessions = new API.Sessions(this);
        this.extensions = new API.Extensions(this);
        this.profiles = new API.Profiles(this);
        this._options = options;
        this.steelAPIKey = steelAPIKey;
    }
    /**
     * Generates a PDF from a specified webpage.
     */
    pdf(body, options) {
        return this.post('/v1/pdf', { body, ...options });
    }
    /**
     * Extracts content from a specified URL.
     */
    scrape(body, options) {
        return this.post('/v1/scrape', { body, ...options });
    }
    /**
     * Captures a screenshot of a specified webpage.
     */
    screenshot(body, options) {
        return this.post('/v1/screenshot', { body, ...options });
    }
    defaultQuery() {
        return this._options.defaultQuery;
    }
    defaultHeaders(opts) {
        return {
            ...super.defaultHeaders(opts),
            ...this._options.defaultHeaders,
        };
    }
    authHeaders(opts) {
        if (this.steelAPIKey == null) {
            return {};
        }
        return { 'steel-api-key': this.steelAPIKey };
    }
}
_a = Steel, _Steel_instances = new WeakSet(), _Steel_baseURLOverridden = function _Steel_baseURLOverridden() {
    return this.baseURL !== 'https://api.steel.dev';
};
Steel.Steel = _a;
Steel.DEFAULT_TIMEOUT = 60000; // 1 minute
Steel.SteelError = Errors.SteelError;
Steel.APIError = Errors.APIError;
Steel.APIConnectionError = Errors.APIConnectionError;
Steel.APIConnectionTimeoutError = Errors.APIConnectionTimeoutError;
Steel.APIUserAbortError = Errors.APIUserAbortError;
Steel.NotFoundError = Errors.NotFoundError;
Steel.ConflictError = Errors.ConflictError;
Steel.RateLimitError = Errors.RateLimitError;
Steel.BadRequestError = Errors.BadRequestError;
Steel.AuthenticationError = Errors.AuthenticationError;
Steel.InternalServerError = Errors.InternalServerError;
Steel.PermissionDeniedError = Errors.PermissionDeniedError;
Steel.UnprocessableEntityError = Errors.UnprocessableEntityError;
Steel.toFile = Uploads.toFile;
Steel.fileFromPath = Uploads.fileFromPath;
Steel.Credentials = Credentials;
Steel.Files = Files;
Steel.Sessions = Sessions;
Steel.SessionslistSessionsSessionsCursor = SessionslistSessionsSessionsCursor;
Steel.Extensions = Extensions;
Steel.Profiles = Profiles;
export { toFile, fileFromPath } from "./uploads.mjs";
export { SteelError, APIError, APIConnectionError, APIConnectionTimeoutError, APIUserAbortError, NotFoundError, ConflictError, RateLimitError, BadRequestError, AuthenticationError, InternalServerError, PermissionDeniedError, UnprocessableEntityError, } from "./error.mjs";
export default Steel;
