// steel-sdk/index.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var _Steel_instances, _a, _Steel_baseURLOverridden;
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnprocessableEntityError = exports.PermissionDeniedError = exports.InternalServerError = exports.AuthenticationError = exports.BadRequestError = exports.RateLimitError = exports.ConflictError = exports.NotFoundError = exports.APIUserAbortError = exports.APIConnectionTimeoutError = exports.APIConnectionError = exports.APIError = exports.SteelError = exports.fileFromPath = exports.toFile = exports.Steel = void 0;
const Core = __importStar(require("./core.js"));
const Errors = __importStar(require("./error.js"));
const Pagination = __importStar(require("./pagination.js"));
const Uploads = __importStar(require("./uploads.js"));
const API = __importStar(require("./resources/index.js"));
const credentials_1 = require("./resources/credentials.js");
const extensions_1 = require("./resources/extensions.js");
const files_1 = require("./resources/files.js");
const profiles_1 = require("./resources/profiles.js");
const sessions_1 = require("./resources/sessions/sessions.js");
/**
 * API Client for interfacing with the Steel API.
 */
class Steel extends Core.APIClient {
    /**
     * API Client for interfacing with the Steel API.
     *
     * @param {string | null | undefined} [opts.steelAPIKey=process.env['STEEL_API_KEY'] ?? null]
     * @param {string} [opts.baseURL=process.env['STEEL_BASE_URL'] ?? https://api.steel.dev] - Override the default base URL for the API.
     * @param {number} [opts.timeout=1 minute] - The maximum amount of time (in milliseconds) the client will wait for a response before timing out.
     * @param {number} [opts.httpAgent] - An HTTP agent used to manage HTTP(s) connections.
     * @param {Core.Fetch} [opts.fetch] - Specify a custom `fetch` function implementation.
     * @param {number} [opts.maxRetries=2] - The maximum number of times the client will retry a request.
     * @param {Core.Headers} opts.defaultHeaders - Default headers to include with every request to the API.
     * @param {Core.DefaultQuery} opts.defaultQuery - Default query parameters to include with every request to the API.
     */
    constructor({ baseURL = Core.readEnv('STEEL_BASE_URL'), steelAPIKey = Core.readEnv('STEEL_API_KEY') ?? null, ...opts } = {}) {
        const options = {
            steelAPIKey,
            ...opts,
            baseURL: baseURL || `https://api.steel.dev`,
        };
        super({
            baseURL: options.baseURL,
            baseURLOverridden: baseURL ? baseURL !== 'https://api.steel.dev' : false,
            timeout: options.timeout ?? 60000 /* 1 minute */,
            httpAgent: options.httpAgent,
            maxRetries: options.maxRetries,
            fetch: options.fetch,
        });
        _Steel_instances.add(this);
        this.credentials = new API.Credentials(this);
        this.files = new API.Files(this);
        this.sessions = new API.Sessions(this);
        this.extensions = new API.Extensions(this);
        this.profiles = new API.Profiles(this);
        this._options = options;
        this.steelAPIKey = steelAPIKey;
    }
    /**
     * Generates a PDF from a specified webpage.
     */
    pdf(body, options) {
        return this.post('/v1/pdf', { body, ...options });
    }
    /**
     * Extracts content from a specified URL.
     */
    scrape(body, options) {
        return this.post('/v1/scrape', { body, ...options });
    }
    /**
     * Captures a screenshot of a specified webpage.
     */
    screenshot(body, options) {
        return this.post('/v1/screenshot', { body, ...options });
    }
    defaultQuery() {
        return this._options.defaultQuery;
    }
    defaultHeaders(opts) {
        return {
            ...super.defaultHeaders(opts),
            ...this._options.defaultHeaders,
        };
    }
    authHeaders(opts) {
        if (this.steelAPIKey == null) {
            return {};
        }
        return { 'steel-api-key': this.steelAPIKey };
    }
}
exports.Steel = Steel;
_a = Steel, _Steel_instances = new WeakSet(), _Steel_baseURLOverridden = function _Steel_baseURLOverridden() {
    return this.baseURL !== 'https://api.steel.dev';
};
Steel.Steel = _a;
Steel.DEFAULT_TIMEOUT = 60000; // 1 minute
Steel.SteelError = Errors.SteelError;
Steel.APIError = Errors.APIError;
Steel.APIConnectionError = Errors.APIConnectionError;
Steel.APIConnectionTimeoutError = Errors.APIConnectionTimeoutError;
Steel.APIUserAbortError = Errors.APIUserAbortError;
Steel.NotFoundError = Errors.NotFoundError;
Steel.ConflictError = Errors.ConflictError;
Steel.RateLimitError = Errors.RateLimitError;
Steel.BadRequestError = Errors.BadRequestError;
Steel.AuthenticationError = Errors.AuthenticationError;
Steel.InternalServerError = Errors.InternalServerError;
Steel.PermissionDeniedError = Errors.PermissionDeniedError;
Steel.UnprocessableEntityError = Errors.UnprocessableEntityError;
Steel.toFile = Uploads.toFile;
Steel.fileFromPath = Uploads.fileFromPath;
Steel.Credentials = credentials_1.Credentials;
Steel.Files = files_1.Files;
Steel.Sessions = sessions_1.Sessions;
Steel.SessionslistSessionsSessionsCursor = sessions_1.SessionslistSessionsSessionsCursor;
Steel.Extensions = extensions_1.Extensions;
Steel.Profiles = profiles_1.Profiles;
var uploads_1 = require("./uploads.js");
Object.defineProperty(exports, "toFile", { enumerable: true, get: function () { return uploads_1.toFile; } });
Object.defineProperty(exports, "fileFromPath", { enumerable: true, get: function () { return uploads_1.fileFromPath; } });
var error_1 = require("./error.js");
Object.defineProperty(exports, "SteelError", { enumerable: true, get: function () { return error_1.SteelError; } });
Object.defineProperty(exports, "APIError", { enumerable: true, get: function () { return error_1.APIError; } });
Object.defineProperty(exports, "APIConnectionError", { enumerable: true, get: function () { return error_1.APIConnectionError; } });
Object.defineProperty(exports, "APIConnectionTimeoutError", { enumerable: true, get: function () { return error_1.APIConnectionTimeoutError; } });
Object.defineProperty(exports, "APIUserAbortError", { enumerable: true, get: function () { return error_1.APIUserAbortError; } });
Object.defineProperty(exports, "NotFoundError", { enumerable: true, get: function () { return error_1.NotFoundError; } });
Object.defineProperty(exports, "ConflictError", { enumerable: true, get: function () { return error_1.ConflictError; } });
Object.defineProperty(exports, "RateLimitError", { enumerable: true, get: function () { return error_1.RateLimitError; } });
Object.defineProperty(exports, "BadRequestError", { enumerable: true, get: function () { return error_1.BadRequestError; } });
Object.defineProperty(exports, "AuthenticationError", { enumerable: true, get: function () { return error_1.AuthenticationError; } });
Object.defineProperty(exports, "InternalServerError", { enumerable: true, get: function () { return error_1.InternalServerError; } });
Object.defineProperty(exports, "PermissionDeniedError", { enumerable: true, get: function () { return error_1.PermissionDeniedError; } });
Object.defineProperty(exports, "UnprocessableEntityError", { enumerable: true, get: function () { return error_1.UnprocessableEntityError; } });
exports = module.exports = Steel;
exports.default = Steel;
